source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs

```diff
@@ -29,44 +34,99 @@ public PollingTentacleBuilder WithTentaclePollSubscription(Uri tentaclePollSubsc
             return this;
         }
 
+        public PollingTentacleBuilder WithCertificate(string CertificatePfxPath, string TentacleThumbprint)
+        {
+            this.CertificatePfxPath = CertificatePfxPath;
+            this.TentacleThumbprint = TentacleThumbprint;
+            return this;
+        }
+
         public PollingTentacleBuilder WithTentacleExe(string tentacleExe)
         {
             tentacleExePath = tentacleExe;
             return this;
         }
 
-        public (IDisposable, Task) Build(CancellationToken cancellationToken)
+        internal RunningTestTentacle Build(CancellationToken cancellationToken)
         {
             var tempDirectory = new TemporaryDirectory();
             var instanceName = Guid.NewGuid().ToString("N");
             var configFilePath = Path.Combine(tempDirectory.DirectoryPath, instanceName + ".cfg");
             var tentacleExe = this.tentacleExePath ?? TentacleExeFinder.FindTentacleExe();
-
+            var subscriptionId = tentaclePollSubscriptionId ?? PollingSubscriptionId.Generate();
+            var CertificatePfxPath = this.CertificatePfxPath ?? Certificates.TentaclePfxPath;
+            var tentacleThumbprint = this.TentacleThumbprint ?? Certificates.TentaclePublicThumbprint;
             CreateInstance(tentacleExe, configFilePath, instanceName, tempDirectory, cancellationToken);
-            AddCertificateToTentacle(tentacleExe, configFilePath, instanceName, Certificates.TentaclePfxPath, tempDirectory, cancellationToken);
+            AddCertificateToTentacle(tentacleExe, configFilePath, instanceName, CertificatePfxPath, tempDirectory, cancellationToken);
             ConfigureTentacleToPollOctopusServer(
                 configFilePath,
                 octopusHalibutPort,
                 octopusThumbprint ?? Certificates.ServerPublicThumbprint,
-                tentaclePollSubscriptionId ?? PollingSubscriptionId.Generate());
+                subscriptionId);
+
+            CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
Why do we need to create a new source if we've already been passed a token? Is it about stack traces upon cancellation or something?